fe/fe-core/src/main/java/org/apache/doris/nereids/trees/AbstractTreeNode.java

```diff
@@ -33,15 +36,43 @@
 
     protected final NodeType type;
     protected final List<TreeNode> children;
+    protected final GroupExpression groupExpression;
+
 
     public AbstractTreeNode(NodeType type, TreeNode... children) {
-        this.type = type;
-        this.children = ImmutableList.copyOf(children);
+        this(type, null, children);
     }
 
-    public AbstractTreeNode(NodeType type, List<NODE_TYPE> children) {
+    /**
+     * Constructor for plan node.
+     *
+     * @param type node type
+     * @param groupExpression group expression related to the operator of this node
+     * @param children children of this node
+     */
+    public AbstractTreeNode(NodeType type, GroupExpression groupExpression, TreeNode... children) {
         this.type = type;
-        this.children = ImmutableList.copyOf(children);
+        if (children.length != 0 && children[0] == null) {
```

Comment:
The special handling of null is too tricky, we should use an dummy Plan denote empty children or don't care children for temporary

Reply:
add a TODO

Reply:
done